lib/ocpp/v16/charge_point_impl.cpp

```diff
@@ -4471,6 +4471,21 @@ void ChargePointImpl::on_firmware_update_status_notification(int32_t request_id,
     if (firmware_update_status == FirmwareStatusNotification::Installed) {
         this->securityEventNotification(ocpp::security_events::FIRMWARE_UPDATED,
                                         std::optional<CiString<255>>("Firmware update was installed"), true);
+    } else if (firmware_update_status == FirmwareStatusNotification::InstallationFailed) {
+        // Restore connector status, since we did not save to db the status, we can just get the old status back using
+        // it
+        auto connector_availability = this->database_handler->get_connector_availability();
+        for (const auto& [connector, availability] : connector_availability) {
```

Comment:
I think we also need to restore the status on connector 0 here. The status of connector 0 is solely controlled by libocpp, not by the libocpp consumer